sovryn-token-bridge/federator/src/lib/Federator.js

```diff
@@ -157,23 +157,55 @@ module.exports = class Federator {
         this.logger.info('get transaction id:', transactionId);
 
         let wasProcessed = await this.federationContract.methods.transactionWasProcessed(transactionId).call();
+        this.logger.info('wasProcessed:', wasProcessed);
         if (!wasProcessed) {
-            let hasVoted = await this.federationContract.methods.hasVoted(transactionId).call({from: from});
-            if (!hasVoted) {
-                this.logger.info(`Voting tx: ${log.transactionHash} block: ${log.blockHash} token: ${symbol}`);
-                await this._voteTransaction(tokenAddress,
-                    receiver,
-                    amount,
-                    symbol,
-                    log.blockHash,
-                    log.transactionHash,
-                    log.logIndex,
-                    decimals,
-                    granularity,
-                    userData
-                );
-            } else {
-                this.logger.debug(`Block: ${log.blockHash} Tx: ${log.transactionHash} token: ${symbol}  has already been voted by us`);
+            this.logger.info('getting transactionIdU:', {
+                tokenAddress,
+                receiver,
+                amount,
+                symbol,
+                blockHash: log.blockHash,
+                transactionHash: log.transactionHash,
+                logIndex: log.logIndex,
+                decimals,
+                granularity,
+                userData
+            });
+
+            let transactionIdU = await this.federationContract.methods.getTransactionIdU(
+                tokenAddress,
+                receiver,
+                amount,
+                symbol,
+                log.blockHash,
+                log.transactionHash,
+                log.logIndex,
+                decimals,
+                granularity,
+                userData || []
```

Comment:
@bravepooh in some cases the userData variable was null, in those cases the tx was reverted so this is to prevent it
the question is how come I get a null userData, perhaps old events don't have that? this needs to be investigated

Reply:
This is True, userData can be null in some cases (when using receiveTokens())